### PR TITLE
Add codecov.io support for test coverage

### PR DESCRIPTION
**Pull request by jdesiloniz**
This PR adds support for code test coverage through codecov.io. Could you please review, @raulraja @dialelo? Thanks a lot!

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,7 +31,21 @@ lazy val formattingSettings = SbtScalariform.scalariformSettings ++ Seq(
       .setPreference(PlaceScaladocAsterisksBeneathSecondAsterisk, true)
 )
 
+lazy val scoverageSettings = Seq(
```
Comment (juanpedromoreno) [inline build.sbt]: Are these settings actually being used?

Comment (jdesiloniz) [inline build.sbt]: Good catch. Removing them as they're not needed (yet). It'd be better to include them when they're required.

**Comment by raulraja** [comment]
LGTM

**Comment by juanpedromoreno** [comment]
LGTM

**Pull request merged**